js/src/workspaces/slot.js

```diff
@@ -171,11 +172,31 @@
 
           _this.eventEmitter.publish('ADD_WINDOW', windowConfig);
 
-        } else if (typeof imageInfoUrl !== 'undefined') {
+        } 
+        
+        else if (typeof imageInfoUrl !== 'undefined') {
           if (!_this.state.getStateProperty('manifests')[imageInfoUrl]) {
             _this.eventEmitter.publish('ADD_MANIFEST_FROM_URL', imageInfoUrl, "(Added from URL)");
           }
-        } else {
+        } 
+        else if (typeof collectionUrl !== 'undefined'){
+          jQuery.getJSON(collectionUrl).done(function (data, status, jqXHR) {
+            if (data.hasOwnProperty('manifests')){
+              jQuery.each(data.manifests, function (ci, mfst) {
+                if (!_this.state.getStateProperty('manifests')[imageInfoUrl]) {
+                  _this.eventEmitter.publish('ADD_MANIFEST_FROM_URL', mfst['@id'], "(Added from URL)");
```

Comment:
I'm loving this new eventEmitter stuff. It's almost like we're a real application.